File: app/src/main/java/com/karkoszka/cookingtime/common/ChooseColorAdapter.java
package com.karkoszka.cookingtime.common;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.karkoszka.cookingtime.R;

public class ChooseColorAdapter extends ArrayAdapter<CTColor> {

	private final Context context;
	  private final CTColor[] values;

	  public ChooseColorAdapter(Context context, CTColor[] values) {
	    super(context, R.layout.listview_item_ct, values);
	    this.context = context;
	    this.values = values;
	  }

	  @Override
	  public View getView(int position, View convertView, ViewGroup parent) {
	    LayoutInflater inflater = (LayoutInflater) context
	        .getSystemService(Context.LAYOUT_INFLATER_SERVICE);
	    View rowView = inflater.inflate(R.layout.listview_item_ct, parent, false);
	    TextView textView = (TextView) rowView.findViewById(R.id.textct1);
	    RelativeLayout  layout = (RelativeLayout) rowView.findViewById(R.id.lay1);
	    layout.setBackgroundColor(values[position].getColor());
	    textView.setText(values[position].getName());
	    Log.d("Color" + position, values[position].getName() + " : " + values[position].getColor());
	    

	    return rowView;
	  }
	 
}

File: app/src/main/java/com/karkoszka/cookingtime/common/ChooseColorAdapter.kt
package com.karkoszka.cookingtime.common

import android.content.Context
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.RelativeLayout
import android.widget.TextView
import com.karkoszka.cookingtime.R

class ChooseColorAdapter(private val context2: Context, private val values: Array<CTColor>) : ArrayAdapter<CTColor?>(context2, R.layout.listview_item_ct, values) {
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        val inflater = context2
                .getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        val rowView = inflater.inflate(R.layout.listview_item_ct, parent, false)
        val textView = rowView.findViewById<View>(R.id.textct1) as TextView
        val layout = rowView.findViewById<View>(R.id.lay1) as RelativeLayout
        layout.setBackgroundColor(values[position].color)
        textView.text = values[position].name
        Log.d("Color$position", values[position].name + " : " + values[position].color)
        return rowView
    }
}